js/whitebit.js

```diff
@@ -1151,6 +1161,42 @@ module.exports = class whitebit extends Exchange {
         };
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        let type = undefined;
+        if (fromAccount === 'main' && toAccount === 'spot') {
+            type = 'deposit';
+        } else if (fromAccount === 'spot' && toAccount === 'main') {
+            type = 'withdraw';
+        }
+        if (type === undefined) {
+            throw new ExchangeError ('This exchange api only allows transfers between main account and spot account');
+        }
+        const request = {
+            'ticker': currency['id'],
+            'method': type,
+            'amount': this.currencyToPrecision (code, amount),
+        };
+        const response = await this.v4PrivatePostMainAccountTransfer (this.extend (request, params));
+        //
+        //    []
+        //
+        const transferOptions = this.safeValue (this.options, 'transfer', {});
+        const fillTransferResponseFromRequest = this.safeValue (transferOptions, 'fillTransferResponseFromRequest', true);
+        const transfer = {
+            'info': response,
+            'status': 'pending',
+        };
+        if (fillTransferResponseFromRequest) {
+            transfer['currency'] = code;
+            transfer['amount'] = amount;
+            transfer['fromAccount'] = fromAccount;
+            transfer['toAccount'] = toAccount;
+        }
+        return transfer;
```

Comment:
Let's move lines 1181-1197 to `parseTransfer`.

Reply:
I agree it's good to have it all in parseTransfer, however so far I haven't been doing it not to change the signature of parseTransfer. What's the best way to pass in amount, fromAccount and toAccount? Should we add like a params argument to parseTransfer? Or add those fields as extra parameters to parseTransfer?

Reply:
Let's make sure that the returned transfer structure is fully unified, including all fields: https://github.com/ccxt/ccxt/wiki/Manual#transfer-structure

Reply:
The point of parseTransfer is to make sure that we return the same unified structure format, so, we can do this (pseudocode):

```JavaScript
parseTransfer (transfer, currency = undefined) { 
    //
    //    []
    //
    return {
        'info': transfer,
        'timestamp': undefined,
        'datetime': undefined,
        // ... other fields of the unified transfer structure ...
    }
}

async transfer (code, amount, ...) {
    await exchange.loadMarkets ();
    const currency = this.currency (code);
    // ...
    const response = await this.privateXXXXXX (...);
    //
    //    []
    //
    const transfer = this.parseTransfer (transfer, currency);
    if (fill...) {
        transfer['toAccount'] = toAccount;
        transfer['fromAccount'] = fromAccount;
    }
    return transfer;
}
```